src/main/java/org/opensearch/agent/tools/SearchAlertsTool.java

```diff
@@ -0,0 +1,177 @@
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        final String tableSortOrder = parameters.getOrDefault("sortOrder", "asc");
+        final String tableSortString = parameters.getOrDefault("sortString", "monitor_name.keyword");
+        final int tableSize = parameters.containsKey("size") && StringUtils.isNumeric(parameters.get("lastUpdateTime"))
+            ? Integer.parseInt(parameters.get("size"))
+            : 20;
+        final int startIndex = parameters.containsKey("startIndex") && StringUtils.isNumeric(parameters.get("lastUpdateTime"))
+            ? Integer.parseInt(parameters.get("startIndex"))
+            : 0;
+        final String searchString = parameters.getOrDefault("searchString", null);
+
+        // not exposing "missing" from the table, using default of null
+        final Table table = new Table(tableSortOrder, tableSortString, null, tableSize, startIndex, searchString);
+
+        final String severityLevel = parameters.getOrDefault("severityLevel", "ALL");
+        final String alertState = parameters.getOrDefault("alertState", "ALL");
+        final String monitorId = parameters.getOrDefault("monitorId", null);
+        final String alertIndex = parameters.getOrDefault("alertIndex", null);
+        @SuppressWarnings("unchecked")
+        final List<String> monitorIds = parameters.containsKey("monitorIds")
+            ? gson.fromJson(parameters.get("monitorIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> workflowIds = parameters.containsKey("workflowIds")
+            ? gson.fromJson(parameters.get("workflowIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> alertIds = parameters.containsKey("alertIds") ? gson.fromJson(parameters.get("alertIds"), List.class) : null;
+
+        GetAlertsRequest getAlertsRequest = new GetAlertsRequest(
+            table,
+            severityLevel,
+            alertState,
+            monitorId,
+            alertIndex,
+            monitorIds,
+            workflowIds,
+            alertIds
+        );
+
+        // create response listener
+        // stringify the response, may change to a standard format in the future
+        ActionListener<GetAlertsResponse> getAlertsListener = ActionListener.<GetAlertsResponse>wrap(response -> {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Alerts=[");
+            for (Alert alert : response.getAlerts()) {
+                sb.append(alert.toString());
+            }
+            sb.append("]");
+            sb.append("TotalAlerts=").append(response.getTotalAlerts());
+            listener.onResponse((T) sb.toString());
+        }, e -> { listener.onFailure(e); });
```

Comment:
Add error log here?

Reply:
Added - covered both alerting tools and AD tool